lib/Semantic_check.ml

```diff
@@ -1,2163 +0,0 @@
-  in
-  let _ =
-    if
-      Core_kernel.String.is_suffix id.name ~suffix:"__"
-      || List.exists
-           (fun str -> str = id.name)
-           [ "true"
-           ; "false"
-           ; "repeat"
-           ; "until"
-           ; "then"
-           ; "var"
-           ; "fvar"
-           ; "STAN_MAJOR"
-           ; "STAN_MINOR"
-           ; "STAN_PATCH"
-           ; "STAN_MATH_MAJOR"
-           ; "STAN_MATH_MINOR"
-           ; "STAN_MATH_PATCH"
-           ; "alignas"
-           ; "alignof"
-           ; "and"
-           ; "and_eq"
-           ; "asm"
-           ; "auto"
-           ; "bitand"
-           ; "bitor"
-           ; "bool"
-           ; "break"
-           ; "case"
-           ; "catch"
-           ; "char"
-           ; "char16_t"
-           ; "char32_t"
-           ; "class"
-           ; "compl"
-           ; "const"
-           ; "constexpr"
-           ; "const_cast"
-           ; "continue"
-           ; "decltype"
-           ; "default"
-           ; "delete"
-           ; "do"
-           ; "double"
-           ; "dynamic_cast"
-           ; "else"
-           ; "enum"
-           ; "explicit"
-           ; "export"
-           ; "extern"
-           ; "false"
-           ; "float"
-           ; "for"
-           ; "friend"
-           ; "goto"
-           ; "if"
-           ; "inline"
-           ; "int"
-           ; "long"
-           ; "mutable"
-           ; "namespace"
-           ; "new"
-           ; "noexcept"
-           ; "not"
-           ; "not_eq"
-           ; "nullptr"
-           ; "operator"
-           ; "or"
-           ; "or_eq"
-           ; "private"
-           ; "protected"
-           ; "public"
-           ; "register"
-           ; "reinterpret_cast"
-           ; "return"
-           ; "short"
-           ; "signed"
-           ; "sizeof"
-           ; "static"
-           ; "static_assert"
-           ; "static_cast"
-           ; "struct"
-           ; "switch"
-           ; "template"
-           ; "this"
-           ; "thread_local"
-           ; "throw"
-           ; "true"
-           ; "try"
-           ; "typedef"
-           ; "typeid"
-           ; "typename"
-           ; "union"
-           ; "unsigned"
-           ; "using"
-           ; "virtual"
-           ; "void"
-           ; "volatile"
-           ; "wchar_t"
-           ; "while"
-           ; "xor"
-           ; "xor_eq" ]
-    then
-      semantic_error ~loc:id.id_loc
-        ("Identifier " ^ id.name ^ " clashes with reserved keyword.")
-  in
-  id
-
-(* Probably nothing to do here *)
-and semantic_check_originblock ob = ob
-
-(* Probably nothing to do here *)
-and semantic_check_returntype = function
-  | Void -> Void
-  | ReturnType ut -> ReturnType (semantic_check_unsizedtype ut)
-
-(* Probably nothing to do here *)
-and semantic_check_unsizedtype = function
-  | Array ut -> Array (semantic_check_unsizedtype ut)
-  | Fun (l, rt) ->
-      Fun
-        ( List.map
-            (function
```

Comment:
could this be `(fun (ob, ut) -> ...`

Reply:
Sure. It means the same thing. I've changed it now.